app/src/main/java/protect/card_locker/ImportExportActivity.java

```diff
@@ -77,6 +77,11 @@ protected void onCreate(Bundle savedInstanceState) {
                     PERMISSIONS_EXTERNAL_STORAGE);
```

Comment:
It probably shouldn't ask for storage permission when a file is shared to it (hmm, wonder if storage permission is even necessary nowadays and if it is which Android versions is is needed for, I need to look into #979 soon...)

Reply:
@TheLastProject,
`    private static final int PERMISSIONS_EXTERNAL_STORAGE = 1;` is just the requestCode.
It is being used here:

```
@Override
    public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);

        if (requestCode == PERMISSIONS_EXTERNAL_STORAGE) {
```

Would you want me to just rename it?

Reply:
My concern is about how the import seems to not care about if the storage permission is granted at all and you get this really weird situation where the import succeeds while there is a dialog asking for storage permission.

On Android 12, this manifest as follows:
https://user-images.githubusercontent.com/1885159/196546608-19def328-1210-4706-968f-735912bcb47b.mp4

On Android 6, the import doesn't seem to do anything at all (the ImportExportActivity starts, but no import is started), which might or might not be related to this. Needs more testing.

I guess this is out of scope, will block a merge but not tagging `hacktoberfest-accepted`. I need to find some time soon to test #979 in every single Android version and figure out exactly which versions need which storage permission when x.x